### PR TITLE
Replace requirejs with webpack to enable ES6+ and non-AMD modules

### PR DESCRIPTION
**Pull request by therewasaguy**
- Build with Webpack instead of requirejs, allowing us to mix AMD with more other/standard module formats (💥  🏅  @oshoham !)
- add babelify so that we can use ES6+
- add `@for <className>` to satisfy the YUIDoc build step in p5.js repo (cc @Spongman, thought you might have insight on this in case there is anything we're missing https://github.com/processing/p5.js-sound/commit/be9410a54fa5119ede1579e3a2337ab943090d15 )

TO DO (perhaps in a separate PR):
- [ ] make we are following best practices for authoring libraries with Webpack https://webpack.js.org/guides/author-libraries

### CODE REVIEW

**Comment by therewasaguy** [comment]
Some relevant issues:

https://github.com/processing/p5.js-sound/issues/228#issuecomment-373515799
#348
#345

File: src/filter.js [hunk 1]
```diff
@@ -257,50 +257,50 @@ define(function (require) {
...
+  //  *  See p5.Filter for methods.
+  //  *
+  //  *  @class p5.LowPass
+  //  *  @constructor
+  //  *  @extends p5.Filter
+  //  */
+  // p5.LowPass = function() {
+  //   p5.Filter.call(this, 'lowpass');
+  // };
+  // p5.LowPass.prototype = Object.create(p5.Filter.prototype);
```
Comment (oshoham) [inline src/filter.js]: @therewasaguy do we want to leave these classes commented out or do we just need to add `@for` to the jsdoc comments?

File: src/filter.js [hunk 2]
```diff
@@ -257,50 +257,50 @@ define(function (require) {
...
+  //  *  See p5.Filter for methods.
+  //  *
+  //  *  @class p5.HighPass
+  //  *  @constructor
+  //  *  @extends p5.Filter
+  //  */
+  // p5.HighPass = function() {
+  //   p5.Filter.call(this, 'highpass');
+  // };
+  // p5.HighPass.prototype = Object.create(p5.Filter.prototype);
```
Comment (oshoham) [inline src/filter.js]: see above comment re: `@for`

File: src/filter.js [hunk 3]
```diff
@@ -257,50 +257,50 @@ define(function (require) {
...
+  //  *  See p5.Filter for methods.
+  //  *
+  //  *  @class p5.BandPass
+  //  *  @constructor
+  //  *  @extends p5.Filter
+  //  */
+  // p5.BandPass = function() {
+  //   p5.Filter.call(this, 'bandpass');
+  // };
+  // p5.BandPass.prototype = Object.create(p5.Filter.prototype);
```
Comment (oshoham) [inline src/filter.js]: see above comment re: `@for`

**Pull request merged**